src/networkd.c

```diff
@@ -1566,6 +1566,12 @@ _netplan_networkd_write_wait_online(const NetplanState* np_state, const char* ro
                               && !(netplan_netdef_get_bond_link(def) || netplan_netdef_get_bridge_link(def)));
             _netplan_address_iter_free(addr_iter);
 
+            // Not all bond members need to be connected (have carrier) for the parent to be ready
+            NetplanNetDefinition* bond_parent = netplan_netdef_get_bond_link(def);
+            if (bond_parent && !d->routable && !d->degraded) {
```

Comment:
suggestion: we could log this message only for members that are not already optional. It would also be nice to identify the interface in the message and that the user is seeing that due to the wait-online configuration.

For example, in my test environment I see the same log message 4 times as I have a bond with 4 interfaces.

Reply:
The log message actually shouldn't trigger for optional bond members, as we have that condition above:

```C
if (!(def->optional || def->activation_mode)) {
```

But mentioning the bond-member name in the log message is a good idea. I'll also add a test for not logging optional interfaces.